File: frontend/src/api/calendar.ts
import { apiFetch } from '../lib/apiClient'
import type { CalendarEvent, CampaignCalendarEventInput } from '../modules/calendar/types'

export function fetchCalendarEvents() {
  return apiFetch<CalendarEvent[]>('/api/calendar/events')
}

export function createCalendarEvent(input: CampaignCalendarEventInput) {
  return apiFetch<CalendarEvent>('/api/calendar/events', {
    method: 'POST',
    body: JSON.stringify({
      date: input.date,
      time: input.time,
      title: input.title,
      campaignId: input.campaignId,
      campaign: input.campaignName,
      place: input.place,
    }),
  })
}

File: backend/internal/models/calendar.go
package models

import "time"

type CalendarEvent struct {
	ID         string    `json:"id"`
	Date       string    `json:"date"` // YYYY-MM-DD
	Time       string    `json:"time,omitempty"`
	Title      string    `json:"title"`
	CampaignID string    `json:"campaignId,omitempty"`
	Campaign   string    `json:"campaign,omitempty"`
	Place      string    `json:"place,omitempty"`
	CreatedBy  string    `json:"createdBy,omitempty"`
	CreatedAt  time.Time `json:"createdAt"`
}

type CreateCalendarEventRequest struct {
	Date       string `json:"date"`
	Time       string `json:"time,omitempty"`
	Title      string `json:"title"`
	CampaignID string `json:"campaignId,omitempty"`
	Campaign   string `json:"campaign,omitempty"`
	Place      string `json:"place,omitempty"`
}
